pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -727,7 +727,7 @@ func (c *Reconciler) createTaskRun(ctx context.Context, rprt *resources.Resolved
 		// We pass the entire, original task ref because it may contain additional references like a Bundle url.
 		tr.Spec.TaskRef = rprt.PipelineTask.TaskRef
 	} else if rprt.ResolvedTaskResources.TaskSpec != nil {
-		tr.Spec.TaskSpec = rprt.ResolvedTaskResources.TaskSpec
+		tr.Spec.TaskSpec = tresources.ApplyPipelineTaskContexts(rprt.ResolvedTaskResources.TaskSpec, rprt.PipelineTask)
```

Comment:
This is throwing me off 😢  Why limit `ApplyPipelineTaskContexts` when a pipelineTask has inlined specification 🤔 

It works great as long as its available as param. Can we please add an example similar to [using_context_variables.yaml
](https://github.com/tektoncd/pipeline/blob/main/examples/v1beta1/pipelineruns/using_context_variables.yaml)


https://github.com/tektoncd/pipeline/blob/01a3388b1d71ff94b421324c93b31d3b55b62a59/examples/v1beta1/pipelineruns/using_context_variables.yaml#L10-L12

I tried using it as a param which fails with:

```
admission webhook "validation.webhook.pipeline.tekton.dev" denied the request: validation failed: non-existent variable in "echo $(params.pipelineTask-retries)\
```